ravenframework/Models/ROM.py

```diff
@@ -172,6 +172,34 @@ def applyRunInfo(self, runInfo):
     """
     self.numThreads = runInfo.get('NumThreads', 1)
 
+  def _manipulateInput(self,dataIn):
+    """
+      Method that is aimed to manipulate the input in order to return a common input understandable by this class
+      @ In, dataIn, object, the object that needs to be manipulated
+      @ Out, inRun, dict, the manipulated input
+    """
+    if len(dataIn)>1:
+      self.raiseAnError(IOError,'Only one input is accepted by the model type '+self.type+' with name'+self.name)
+    inRun = super()._manipulateInput(dataIn)
+    return inRun
+
+  def createNewInput(self,myInput,samplerType,**kwargs):
+    """
+      This function will return a new input to be submitted to the model, it is called by the sampler.
+      here only a PointSet is accepted a local copy of the values is performed.
+      For a PointSet, only the last set of entries are copied
+      The copied values are returned as a dictionary back
+      @ In, myInput, list, the inputs (list) to start from to generate the new one
+      @ In, samplerType, string, is the type of sampler that is calling to generate a new input
+      @ In, **kwargs, dict,  is a dictionary that contains the information coming from the sampler,
+           a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
+      @ Out, ([(inputDict)],copy.deepcopy(kwargs)), tuple, return the new input in a tuple form
+    """
+    if len(myInput)>1:
+      self.raiseAnError(IOError,'Only one input is accepted by the model type '+self.type+' with name'+self.name)
+    [(inputDict)],kwargs = super().createNewInput(myInput,samplerType,**kwargs)
+    return [(inputDict)],copy.deepcopy(kwargs)
```

Comment:
It seems a little bit redundant here, may be you can do:
```Python
     newInput, kwargs = super().createNewInput(myInput,samplerType,**kwargs)
     return newInput, kwargs
```
I think there is no need to do deep copy here, since the deep copy is handled in super class.

If you change the code, please also update the docstring.

Reply:
deepcopy removed, i kept the original format for consistency